tests/test_basic.py

```diff
@@ -9,6 +11,7 @@ def test_heuristic_completion():
     assert output is not None
 
 
+@pytest.mark.skipif(sys.version_info > (2,7), reason="Only runs on Python2.")
```

Comment:
Why we test on py2 instead py3?

Reply:
This pickle doesn't load properly in py3. We need a py3 version for this test to work.

Reply:
Specifically, you'll get this failure in Python3
```
________________________________________________ test_ml_completion _________________________________________________

    def test_ml_completion():
        """Simply test that ML-based parse runs without errors."""
        output = pdftotree.parse("tests/input/paleo.pdf", "tests/output/",
                                 model_path="tests/input/paleo_model.pkl",
>                                debug=True)

tests/test_basic.py:18: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
pdftotree/core.py:57: in parse
    pdf_tree = extractor.get_tree_structure(model, favor_figures)
pdftotree/TreeExtract.py:173: in get_tree_structure
    table_predictions = model.predict(candidates_features)
.venv/lib/python3.5/site-packages/sklearn/linear_model/base.py:324: in predict
    scores = self.decision_function(X)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = LogisticRegression(C=None, class_weight=None, dual=None, fit_intercept=None,
          intercept_scaling=None, max_ite...          n_jobs=None, penalty=None, random_state=None, solver=None,
          tol=None, verbose=None, warm_start=None)
X = array([[1.00000000e+00, 3.20000000e+01, 1.00000000e+00, 4.70000000e+01,
        1.00000000e+00, 1.40000000e+01, 1.0000...73061024e-01, 4.26805070e-01, 8.74078239e-01,
        6.03646597e-01, 0.00000000e+00, 9.80000000e+01, 6.00000000e+00]])

    def decision_function(self, X):
        """Predict confidence scores for samples.
    
            The confidence score for a sample is the signed distance of that
            sample to the hyperplane.
    
            Parameters
            ----------
            X : {array-like, sparse matrix}, shape = (n_samples, n_features)
                Samples.
    
            Returns
            -------
            array, shape=(n_samples,) if n_classes == 2 else (n_samples, n_classes)
                Confidence scores per (sample, class) combination. In the binary
                case, confidence score for self.classes_[1] where >0 means this
                class would be predicted.
            """
        if not hasattr(self, 'coef_') or self.coef_ is None:
            raise NotFittedError("This %(name)s instance is not fitted "
>                                "yet" % {'name': type(self).__name__})
E           sklearn.exceptions.NotFittedError: This LogisticRegression instance is not fitted yet

.venv/lib/python3.5/site-packages/sklearn/linear_model/base.py:298: NotFittedError
----------------------------------------------- Captured stdout call ------------------------------------------------
Loading pretrained model for table detection
Model loaded!
Digitized PDF detected, building tree structure
================================================= warnings summary ==================================================
tests/test_basic.py::test_ml_completion
  /home/lwhsiao/repos/pdftotree/.venv/lib/python3.5/site-packages/sklearn/base.py:311: UserWarning: Trying to unpickle estimator LogisticRegression from version pre-0.18 when using version 0.19.1. This might lead to breaking code or invalid results. Use at your own risk.
    UserWarning)

-- Docs: http://doc.pytest.org/en/latest/warnings.html
```

Reply:
Got it! Will fix later.